crates/red_knot_python_semantic/src/types/call.rs

```diff
@@ -48,6 +48,10 @@ pub(super) enum CallOutcome<'db> {
         binding: CallBinding<'db>,
         asserted_ty: Type<'db>,
     },
+    Cast {
+        binding: CallBinding<'db>,
+        casted_ty: Type<'db>,
+    },
```

Comment:
I didn't complete a review here before it was landed, but I was going to recommend that we not add this `CallOutcome` variant. Put up a follow-up PR at https://github.com/astral-sh/ruff/pull/15461